### DISCUSSION TITLE
Canvas Renderer - destroying an texture that is on stage.

### DISCUSSION THREAD

**Discussion by owendwyer** [pixijs/pixijs#7782] [General]
For various reasons I have to reuse loaders and destroy textures.

When using Canvas Renderer, destroying a texture that is actively being rendered will cause an error.

This example uses v6.1.2/pixi-legacy.js and forceCanvas.  When the flower texture is destroyed, the page crashes and bunny stops spinning.

https://www.pixiplayground.com/#/edit/W-3rc1gJhscWnpOaTuLYJ

The error comes when the sprite renderer looks for the texture frame width, but the texture frame is null.  I think this can be resolved in CanvasSpriteRenderer.ts by moving the if(!texture.valid)return; check so that it comes before looking for the frame width.

https://github.com/pixijs/pixijs/compare/dev...owendwyer:patch-1

**Comment by ivanpopelyshev** [comment]
Make PR?
